.github/workflows/release.yaml

```diff
@@ -0,0 +1,177 @@
+        id: find_pr
+        with:
+          state: all
+      - id: pr_labels
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const issue_number = ${{ steps.find_pr.outputs.pr }}
+            const owner = context.repo.owner
+            const repo = context.repo.repo
+
+            const attached_labels = (await github.rest.issues.listLabelsOnIssue({ owner, repo, issue_number })).data
+
+            const is_release_label_attached = attached_labels.find(label => label.name == 'release') !== undefined
+            core.setOutput('is_release_label_attached', is_release_label_attached)
+  release:
+    runs-on: ubuntu-latest
+    needs: check_label
+    if: needs.check_label.outputs.is_release_label_attached == 'true'
+    env:
+      EARTHLY_TOKEN: "${{ secrets.EARTHLY_TOKEN }}"
+      CARGO_REGISTRY_TOKEN: "${{ secrets.CARGO_REGISTRY_TOKEN }}"
+      FORCE_COLOR: 1
+    steps:
+      - uses: earthly/actions-setup@v1
+        with:
+          version: v0.8.3
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - uses: jwalton/gh-find-current-pr@v1
+        id: find_pr
+        with:
+          state: all
+      - run: npm install toml globby axios
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const is_merge_event = ${{ github.event_name == 'push' }}
+
+            const { globby } = await import("${{ github.workspace }}/node_modules/globby/index.js");
+            const toml = require('toml');
+            const fs = require('fs');
+            const axios = require('axios');
+            const { spawn } = require('child_process');
+
+            async function get_crate_details(name) {
+              try {
+                return (await axios(`https://crates.io/api/v1/crates/${name}`)).data
+              } catch(error) {
+                console.error("Error fetching crate details:", error)
+              }
+            }
+
+            function run_command_stream(command_arr) {
+              const [command, ...args] = command_arr
+
+              const child = spawn(command, args)
+              child.stdout.on("data", data => console.log(Buffer.from(data).toString()))
+              child.stderr.on("data", data => console.error(Buffer.from(data).toString()))
+              child.on("close", code => {
+                console.log(command, "exited with code", code)
+                if (code != 0) {
+                  core.setFailed(`Command ${command_str} exited with code ${code}`)
+                }
+              })
+            }
+
+            async function create_or_update_comment(identifier, comment_body) {
+              const comment_header = `<!-- ${identifier} -->\n`
+              const comment = comment_header + comment_body
+
+              const issue_number = ${{ steps.find_pr.outputs.pr }}
```

Comment:
I think that depends on whether we care more about avoiding repetition or separation of concerns here. 

In this case, I don't think repeating the `find_pr` step is too concerning, as the code complexity and run-time cost are low. With that, I would want to keep the `check_label` step doing just one thing - check for labels.

What do you think?

Reply:
To me, it looks outputting the PR number from check_label makes sense, since  `check_label` job's label_attached output is specific to a PR.
label_attached without PR number lacks context.

Reply:
Addressed in 3d26d56